### PR TITLE
Logical AND / OR fix with error input.

### PR DESCRIPTION
**Pull request by TristonianJones**
The google/cel-spec defines the logical AND / OR operators as commutative and capable
of absorbing errors. The interpreters do not handle this case correctly and will in some
scenarios produce a faulty result. In the majority of cases the bug is a non-issue, but is out
of conformance with the spec.

Once this test is submitted a conformance test will be written to exercise this case and a point
release created.

### CODE REVIEW

File: interpreter/interpretable.go [hunk 1]
```diff
@@ -177,8 +177,11 @@ func (or *evalOr) Eval(ctx Activation) ref.Val {
 	if types.IsUnknown(rVal) {
 		return rVal
 	}
-	// if the left-hand side is non-boolean return it as the error.
-	return types.ValOrErr(lVal, "no such overload")
+	// If the left-hand side is non-boolean return it as the error.
+	if types.IsError(lVal) {
+		return types.ValOrErr(lVal, "no such overload")
```
Comment (JimLarson) [inline interpreter/interpretable.go]: Just return lVal here, since it's known to be an error.

File: interpreter/interpretable.go [hunk 2]
```diff
@@ -218,8 +221,11 @@ func (and *evalAnd) Eval(ctx Activation) ref.Val {
...
```
Comment (JimLarson) [inline interpreter/interpretable.go]: Just return lVal here, since it's known to be an error.

File: interpreter/interpretable.go [hunk 3]
```diff
@@ -632,7 +638,11 @@ func (or *evalExhaustiveOr) Eval(ctx Activation) ref.Val {
 	if types.IsUnknown(rVal) {
 		return rVal
 	}
-	return types.ValOrErr(lVal, "no such overload")
+	// If the left-hand side is non-boolean return it as the error.
+	if types.IsError(lVal) {
+		return types.ValOrErr(lVal, "no such overload")
```
Comment (JimLarson) [inline interpreter/interpretable.go]: Just return lVal.

File: interpreter/interpretable.go [hunk 4]
```diff
@@ -668,7 +678,11 @@ func (and *evalExhaustiveAnd) Eval(ctx Activation) ref.Val {
...
```
Comment (JimLarson) [inline interpreter/interpretable.go]: Just return lVal.

**Review by JimLarson — APPROVED** [review]

**Comment by TristonianJones** [comment]
Correction, the right type of response was being given, but the wrong error might have been returned before the change.

**Pull request merged**